cores/esp8266/Updater.cpp

```diff
@@ -463,14 +464,15 @@ size_t UpdaterClass::writeStream(Stream &data) {
         }
         toRead = data.readBytes(_buffer + _bufferLen,  bytesToRead);
         if(toRead == 0) { //Timeout
-            delay(100);
-            toRead = data.readBytes(_buffer + _bufferLen, bytesToRead);
-            if(toRead == 0) { //Timeout
-                _currentAddress = (_startAddress + _size);
-                _setError(UPDATE_ERROR_STREAM);
-                _reset();
-                return written;
-            }
+          if (millis() - timeout > streamTimeout) {
```

Comment:
Is this valid when `millis()` just overflows between calls?

Reply:
As I said, pkease reimplement the timeout in terms of the polledTimeout class.

Reply:
I do believe it should be fine during rollovers. https://arduino.stackexchange.com/questions/12587/how-can-i-handle-the-millis-rollover
Is that right?